packages/mdc-shape/_functions.scss

```diff
@@ -114,15 +87,27 @@
       @error "Invalid radius: '#{$radius}' is more than 4 values";
     }
 
-    $radius-value: ();
+    $radius-values: ();
 
-    @each $corner in $radius {
-      $radius-value: append($radius-value, mdc-shape-prop-corner-value_($corner));
+    @for $i from 1 through length($radius) {
+      $corner: nth($radius, $i);
+
+      @if map-has-key($mdc-shape-category-values, $corner) {
```

Comment:
What happens if `small` value is `4px 0 4px 0` and I apply this mixin with this value - `mdc-button-shape-radius(small 0 small 0)`?

I'm OK not supporting something like `mdc-button-shape-radius(small 0 small 0)` if that makes the implementation complex.

Reply:
That would result in `4px 0 4px 0`. It maps each respective corner from the shape category.